src/shared/components/comment/comment-node.tsx

```diff
@@ -849,14 +850,30 @@ export class CommentNode extends Component<CommentNodeProps, CommentNodeState> {
 
   linkBtn(small = false) {
     let cv = this.props.node.comment_view;
+    let classnames = classNames("btn btn-link btn-animate text-muted", {
+      "btn-sm": small,
+    });
+
+    let title = this.props.showContext
+      ? i18n.t("show_context")
+      : i18n.t("link");
+
     return (
-      <Link
-        className={`btn ${small && "btn-sm"} btn-link btn-animate text-muted`}
-        to={`/post/${cv.post.id}/comment/${cv.comment.id}`}
-        title={this.props.showContext ? i18n.t("show_context") : i18n.t("link")}
-      >
-        <Icon icon="link" classes="icon-inline" />
-      </Link>
+      <>
+        <Link
+          className={classnames}
+          to={`/post/${cv.post.id}/comment/${cv.comment.id}`}
+          title={title}
+        >
+          <Icon icon="link" classes="icon-inline" />
+        </Link>
+        {/* TODO comment ap_ids are currently broken anyway, so use post.ap_id, and wait until comment tree / endpoint refactor */}
+        {!cv.comment.local && (
+          <a className={classnames} title={title} href={cv.post.ap_id}>
+            <Icon icon="fedilink" classes="icon-inline" />
+          </a>
```

Comment:
It should probably be shown for local comments/posts as well, otherwise it will also be confusing that some dont have it.

Reply:
In fact, i dont think it makes sense to show separate icons for link and activitypub link. If you want to share a specific post/comment with someone, it should be using the link where it was originally posted, not on the instance where you happen to browse. So it should show the ap_id in all cases (not sure which icon though). 

That means we definitely have to implement a view under `/post/123/comment/456`, but that should be really easy if its simply a redirect to `/comment/456`.

Reply:
> It should probably be shown for local comments/posts as well, otherwise it will also be confusing that some dont have it.

In that case it'd just be the same link twice, which would be pointless. The clutter will start to pile up on mobile especially. 

> If you want to share a specific post/comment with someone, it should be using the link where it was originally posted, not on the instance where you happen to browse. So it should show the ap_id in all cases (not sure which icon though).

Not sure abt this one. I could see some cases where you'd want to link a specific instance's cached version of that post, esp if one instance goes down, gets blocked, or you're linking to someone you know doesn't have an account on the original. But I do agree in most cases the federated one makes sense. Maybe for federated / non-local content, just don't show the original link?

> That means we definitely have to implement a view under /post/123/comment/456, but that should be really easy if its simply an alias for /comment/456

The actual comment `ap_id` is just `instance/comment/number`, which doesn't include the post. The `/post/number/comment/number` is a front end scrolling hack that I'm starting to regret adding, because its really not the right way to refer to comments. I'd rather this just wait on the proper comment endpoint than continue working building on the hack.